pkg/sqlutil/sqltest/data_source.go

```diff
@@ -0,0 +1,65 @@
+package sqltest
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+)
+
+// NewNoOpDataSource returns an empty DataSource type which will satisfy the interface
+func NewNoOpDataSource() sqlutil.DataSource {
+	return &noOpDataSource{}
+}
+
+type noOpDataSource struct{}
```

Comment:
Sorry, is it clear why you cannot use dataSource from pkg/sqlutil/hook_test.go instead of noOpDataSource ?

Reply:
From what I understand noOpDataSource is intended to be used in some tests. Would dataSource from pkg/sqlutil/hook_test.go just work as it is now? (Or alternatively, one could also call this HookableDataSource that can be initialized with hookBehaviour = false / true)

Reply:
> Would dataSource from pkg/sqlutil/hook_test.go just work as it is now?

I don't think that an exported "no op" type should be arbitrarily sleeping for any amount of time.

> (Or alternatively, one could also call this HookableDataSource that can be initialized with hookBehaviour = false / true [this is just another way to implement the desire behavior without embedding])

Can you elaborate?

Reply:
> > Would dataSource from pkg/sqlutil/hook_test.go just work as it is now?
> 
> I don't think that an exported "no op" type should be arbitrarily sleeping for any amount of time.

I agree, but it would sleep only if GetContext/SelectContext is called - I am not sure how often this would happen.

> 
> > (Or alternatively, one could also call this HookableDataSource that can be initialized with hookBehaviour = false / true [this is just another way to implement the desire behavior without embedding])
> 
> Can you elaborate?

Instead of embedding the common functions and having two structs HookedDataSource and NoOpDataSource, one could have one HookableDataSource that would behave like HookedDataSource if initialized with true and like NoOpDataSource if initialized with false.

Reply:
> Instead of embedding the common functions and having two structs HookedDataSource and NoOpDataSource, one could have one HookableDataSource that would behave like HookedDataSource if initialized with true and like NoOpDataSource if initialized with false.

If there is a way to expose a _general_ type that can be _customized_ to do exactly what the current hook type is doing, then that would be great. But I don't think it would make sense to put this specialized behavior behind a flag.

Reply:
Sorry, do you mean that something like

```
type dataSource struct {
	hookable bool
}

func (q *dataSource) SelectContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error {
	if q.hookable {
		select {
		case <-ctx.Done():
			return ctx.Err()
		case <-time.After(selDur):
		}
	}
	return nil
}
```

is not good?

Reply:
I think that behavior is arbitrary in the context of a general exported type. It is just what one particular set of tests needed at the time, and there is no reason to assume that any other users would want the same behavior (in principle or with those particular const durations).

Reply:
I agree, I am just trying to find a way to avoid code duplication. I would also be fine with the embed solution.